RyanNoelk/OpenEats
# Feature - Grocery List

This PR is a WIP.

@hermzz I have all the core features done for this. I still have a good amount of cleanup work for this, so you don't have to look thought the code (yet :P). But there is enough work done that I want your feedback on the overall flow. You can create/update/delete lists as well create/update/delete items on a list. 

To test:
- Update python requirements
- Run migrations
- npm install
- Log in
- Navigate to My Account -> Grocery List

My goal is that the UI should be simple and easy enough for anyone to use and understand. If you're not understanding anything let me know and I will try and explain/fix it.

Things still to do:
- unit tests for the API
- translations
- ~api code comments~
- react code comments
- ~drop dead code~
- ~clean up css (break it up into separate files)~
- ~add trans tags to all react components~
- ~reduce dom re-loads~
- ~reduce API calls~ 
- ~user validation on API and front-end~
- probably more things I'm not thinking of right now :)
